src/Plot.tsx

```diff
@@ -91,7 +87,11 @@ export default function Plot({
     xFlip: false,
     yFlip: false,
   };
-  const [state, dispatch] = useReducer(reducer, initialState, undefined);
+  const [state, dispatch] = useReducer(
+    produce(reducer),
```

Comment:
Can you keep this call outside of the component to avoid creating a new function on every render?